core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -39,22 +37,22 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
     }
   }
 
-  def starting(work: Option[(Work, ActorRef)], askWorkImmediately: Boolean): Receive = whileWaiting(Starting) orElse {
+  def waitingForRoutee(work: Option[(Work, ActorRef)]): Receive = whileWaiting(Starting) orElse {
```

Comment:
This change isn't as safe as you think. The routee could die when it's waiting for a work from the queue, that is,  it's already asked for work (scenario in line 74). In this case you shouldn't ask again (although it might be fine since the extra work would probably be rejected). I'll leave it to you to fix it now or just "fix it" by removing the whole `waitingForRoutee` state